docs/data/tree-view/simple-tree-view/customization/GmailTreeView.js

```diff
@@ -77,15 +82,19 @@ const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
     status,
   } = useTreeItem({ id, itemId, children, label, disabled, rootRef: ref });
 
-  const style = {
-    '--tree-view-color': theme.palette.mode !== 'dark' ? color : colorForDarkMode,
```

Comment:
This solution could still work if `theme.palette.mode` returned the actual mode set on docs... 🙈 🤷

Reply:
The idea is to delegate the light/dark to CSS only. Otherwise, when you update the theme mode all the Demo ThemeProvider will re-render.

The new Demo theme provider that erase custom theme tokens:

https://github.com/mui/material-ui/pull/45386/files#diff-7527a515d20eff239825c53744ce9602043624db93e0a7275943c9fa95256bfbR32-R45

If some packages plan to have demo that do not support CSS-vars we could have a dedicated Demo wrapper that pass the mode token with something like

```
{{"demo": "NeedThemeToken.js", "notCSSVarsCompatible": true }}
```